design-proposals/volume-migration/volume-migration.md

```diff
@@ -0,0 +1,466 @@
+
+At the moment, PVCs can only be migrated using this API if they belong to a single running virtual machine. Consequently, all volume migration objects that have migrated volumes that are associated with different VMIs will be rejected by the controller. The object is in the `Pending` state if none of the migrated volumes have been assigned to a VMI, since the VMI might be created at a later time.
+
+The `VolumeMigration` expects a list of source and destination PVCs and the destination PVCs needs to be created either manually by the user or other tools.
+
+According to the `ReclaimPolicySourcePvc`, KubeVirt will determine how to handle the source PVCs. The source PVC will be automatically deleted under the `Delete` policy, but it will remain untouched under the `Retain` policy.
+
+The core idea is that it is KubeVirt responsibility to handle the lifecycle of the source PVCs as it is the only component able to orchestrate and set the migration as successful.
+
+Certain types of volumes and disks aren't suitable to be migrated or the feature isn't supported yet. Those migrated volumes will be marked as rejected. 
+If there are other valid volumes belonging to the same VMI as the rejected ones, the VM migration won't be created for those volumes as well. Those volumes will be listed as `Valid` in the `VolumeMigrationStates` field.
+
+The current types of rejected volumes are:
+  * Hotpluggable volumes, this case will be considered in a later iteration.
+  * Filesystem volumes, currently virtiofs doesn't support live-migration.
+  * Shareable volumes, if a disk is shared between multiple VMs, it cannot be safely copied.
+  * LUNs disks, originally the disk might support SCSI protocol but the destination PVC class not. This case will be considered in a later iteration.
+
+In the `VolumeMigrationStatus` reports the phase of the migration:
+  * `Pending` if there are no VMI associated to the migrated volumes
+  * `Scheduling` if the VMI migration object has been created but the migration hasn't started yet
+  * `Running` once the VMI migration starts
+  * `Succeeded` if the VMI migration has succeeded
+  * `Failed` if the VMI migration has failed or there are rejected volumes
+
+## Design details and motivation
+
+### Motivation on the approaches
+
+From libvirt side there a multiple ways to achieve the storage migration. It is possible to simply copy the storage on the new destination and then swap the original disk wit the new one. This approach is the most efficient as the overhead is caused only by the storage copy.
+
+A different approach is to simultaneously move the storage and live-migrate the guest. When the destination storage is not available on the host where the VM is executing, this solution is required. For this case, costs are greater since there is a storage as well as a live VM migration overhead.
```

Comment:
```suggestion
A different approach is to simultaneously move the storage and live-migrate the guest. When the destination storage is not available on the host where the VM is executing, this solution is required. For this case, costs are greater since there is storage as well as live VM migration overhead.
```